src/pgduckdb_hooks.cpp

```diff
@@ -169,10 +182,16 @@ IsAllowedStatement(Query *query, bool throw_error) {
 	/* We don't support modifying statements on Postgres tables yet */
 	if (query->commandType != CMD_SELECT) {
 		RangeTblEntry *resultRte = list_nth_node(RangeTblEntry, query->rtable, query->resultRelation - 1);
-		if (!::IsDuckdbTable(resultRte->relid)) {
+		if (::IsHeapTable(resultRte->relid) && query->commandType == CMD_INSERT) {
+			if (query->onConflict || query->returningList) {
+				elog(elevel, "DuckDB does not support INSERT ON CONFLICT or RETURNING on Postgres tables");
+				return false;
+			}
```

Comment:
Why not allow these? Do they not work? I would expect that if we insert using postgres its functionality that thees things would automatically work.

Reply:
To be clear, not saying that this should be implemented in the first version of this feature. But I'd at least like a code comment explaining why we do not support it.

Reply:
Ah, block them since I haven't tested them yet. I've done some initial testing, and RETURNING works out of box while onConflict doesn't. Will dig deeper into the WHY and add comments to clarify.

Reply:
ON CONFLICT is also working